nntrainer/graph/network_graph.cpp

```diff
@@ -687,27 +688,59 @@ bool NetworkGraph::canExecuteInPlace(const std::shared_ptr<LayerNode> &lnode) {
     };
 
   /**
+   * @note Conditions to decide if this layer node can be in-place:
    * 1. if the layer is a no-op, then it can operate in-place as it is not
    * modifying its input/output tensors and does not need to check its
    * neighboring nodes for dependency.
    * 2. if the layer is not supporting backwarding, there is no dependency
    * requirement with other nodes for backwarding.
+   *
+   * @note Conditions to decide the type of inplace for this layer:
+   * 1. if the previous layers were restricting, then this layer will also be
+   * restricting.
```

Comment:
It might be good if you give some examples here. the previous layer is restricted means that the previous layer is in-place layer. so.. the next layer of in place layer will be in place as well?

Reply:
This was part of the previous PR, where it was expanded upon.